src/zcash/JoinSplit.hpp

```diff
@@ -59,7 +83,7 @@ class JoinSplit {
                          const uint256& pubKeyHash
                         );
 
-    virtual ZCProof prove(
+    virtual JSProofWitness<NumInputs, NumOutputs> witness(
```

Comment:
wondering if just `JSWitness` is better. `JSProofWitness` sounds like the actual QAP witness.